lua/neogit/status.lua

```diff
@@ -638,18 +638,18 @@ local cmd_func_map = function ()
         status_buffer:close()
       end, 0)
     end,
-    ["Depth1"] = function()
-      set_folds({ true, true, false })
-    end,
-    ["Depth2"] = function()
-      set_folds({ false, true, false })
-    end,
-    ["Depth3"] = function()
-      set_folds({ false, false, true })
-    end,
-    ["Depth4"] = function()
-      set_folds({ false, false, false })
-    end,
+    ["Depth1"] = void(async(function()
+      await(set_folds({ true, true, false }))
+    end)),
+    ["Depth2"] = void(async(function()
+      await(set_folds({ false, true, false }))
+    end)),
+    ["Depth3"] = void(async(function()
+      await(set_folds({ false, false, true }))
+    end)),
+    ["Depth4"] = void(async(function()
+      await(set_folds({ false, false, false }))
+    end)),
```

Comment:
This shouldn't change anything afaik. Am I missing something?

Reply:
If I don't wrap these functions in async I get the c-bound error.

Reply:
Aaah and it needs to be async because i made the refresh async and forgot to make the `set_folds` method async too. I was missing something quite big, apparently :sweat_smile: